### PR TITLE
Fix denominator constraint

### PR DESCRIPTION
**Pull request by filiplazovic**
Fixed a bug where user was able to pass denominator smaller than the correct value, therefore maximizing his reward and draining reward pot.

### CODE REVIEW

File: gasCosts/gasCosts.js [hunk 1]
```diff
@@ -313,7 +313,7 @@ contract("All", accounts => {
     });
 
     it("when working with reward payouts", async () => {
-      const totalReputation = toBN(350 * 1e18);
+      const totalReputation = toBN(300 * 1e18);
```
Comment (area) [inline gasCosts/gasCosts.js]: You've changed this to be accurate, but I don't think this variable is used anywhere in the test?

File: helpers/test-helper.js [hunk 2]
```diff
@@ -253,5 +253,9 @@ export function bnSqrt(bn) {
       .add(a)
       .div(web3Utils.toBN(2));
   }
+  // Do we need while loop? Not sure how precise the formula above is
```
Comment (area) [inline helpers/test-helper.js]: I don't believe the loop is needed (though obviously the addition is). I believe you could also replace the addition with `b = b.addn(1)`.

File: test/colony-funding.js [hunk 3]
```diff
@@ -812,24 +799,28 @@ contract("Colony Funding", addresses => {
         "colony-reward-payout-invalid-parametar-user-token",
         "colony-reward-payout-invalid-parametar-total-reputation",
         "colony-reward-payout-invalid-parametar-total-tokens",
-        "colony-reward-payout-invalid-parametar-amount",
         "colony-reward-payout-invalid-parametar-numerator",
-        "colony-reward-payout-invalid-parametar-denominator"
+        "colony-reward-payout-invalid-parametar-denominator",
```
Comment (area) [inline test/colony-funding.js]: It's spelled `parameter` 😄

File: test/colony-funding.js [hunk 4]
```diff
@@ -842,28 +833,72 @@ contract("Colony Funding", addresses => {
...
+
+    it("should return correct info about reward payout", async () => {
+      const { logs } = await colony.startNextRewardPayout(otherToken.address);
+      const payoutId = logs[0].args.id;
+
+      const balance = await colony.getPotBalance(0, otherToken.address);
+      const blockTimestamp = await currentBlockTime();
+
+      const info = await colony.getRewardPayoutInfo(payoutId);
+      assert.equal(info[1].toString(), userTokens.toString());
```
Comment (area) [inline test/colony-funding.js]: Is there a reason we're not checking `info[0]`? We could compare it to `colonyNetwork.getReputationRootHash()`, right? (Although without going through all the mining setup, I suppose that's going to just be `0x0`, which might be the reason you've skipped it?)

Comment (filiplazovic) [reply test/colony-funding.js]: Yea, i thought i can add it when we add proofs (which will require mining process), but i guess we can do it now

File: gasCosts/gasCosts.js (hunk 1 above)
Comment (filiplazovic) [reply gasCosts/gasCosts.js]: We are passing it to `claimRewardPayout`

**Pull request merged**